specs/altair/beacon-chain.md

```diff
@@ -511,16 +511,29 @@ def apply_deposit(state: BeaconState,
         signing_root = compute_signing_root(deposit_message, domain)
         # Initialize validator if the deposit signature is valid
         if bls.Verify(pubkey, signing_root, signature):
-            state.validators.append(get_validator_from_deposit(pubkey, withdrawal_credentials, amount))
-            state.balances.append(amount)
+            index = get_index_for_new_validator(state)
```

Comment:
Suggesting to define `add_validator_to_registry` (there could be a better name) and doing the following refactor:

### Phase0
```python
def add_validator_to_registry(state: BeaconState, validator: Validator) -> None:
    # Add validator and balance entries
    state.validators.append(get_validator_from_deposit(pubkey, withdrawal_credentials, amount))
    state.balances.append(amount)
```

Replace the two lines with a call to this method

### Altair

Modify `add_validator_to_registry` to be aligned with Altair updates. Which also makes Altair spec leaner as a side effect (no need to redefine entire `apply_deposit`, just `add_validator_to_registry`).

### Reuse indexes
Modify `add_validator_to_registry` respectively, and encapsulate `get_index_for_new_validator` and update or append logic in this feature only.

Reply:
The intent of `update_or_append_to_list` was to prevent having to duplicate initialization statements, which will keep growing with every fork. No strong opinion on either way. I'll PR the `add_validator_to_registry` separately since it's mostly orthogonal to the change